### PR TITLE
add --app-version flag to 'helm package'

### LINKED ISSUE #2566: Specifying appVersion with "helm package"
As I'd like to generate a helm package at build time, I would like to specify the ``appVersion`` as a command line parameter of the ``helm package`` command, just like you can do with the chart version.

Or is there an alternative way to do that? (I'm totally new to helm)

### PR DESCRIPTION
**Pull request by adshmh**
Add the option 'app-version' to helm package command. When 'helm package --app-version foo' is run, this will set the AppVersion field to 'foo' in the packaged chart. Fixes #2566 

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

### CODE REVIEW

**Comment by bacongobbler** [comment]
Hey @adshmh! Thanks for the PR!

It looks like CI failed on this PR. Would you mind fixing up the tests?

> helm docs are out of date. Please run "make docs"

**Comment by adshmh** [comment]
@bacongobbler Thanks for the reminder. Fixed the doc issue.

File: cmd/helm/package.go [hunk 1]
```diff
@@ -139,6 +141,11 @@ func (p *packageCmd) run() error {
 		debug("Setting version to %s", p.version)
 	}
 
+	if len(p.appVersion) != 0 {
```
Comment (bacongobbler) [inline cmd/helm/package.go]: it's probably simpler to just check `if p.appVersion != ""` instead of casting to an int.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by bacongobbler** [comment]
one little change and it looks good to go!

File: cmd/helm/package.go (hunk 1 above)
Comment (adshmh) [reply cmd/helm/package.go]: Thank you for the review. I have updated the PR.

**Review by bacongobbler — APPROVED** [review]

**Comment by asetty** [comment]
On a similar note, it would be nice to have an --app-version flag for helm upgrade / helm install.
I tried passing my desired appVersion with `helm upgrade --set` but learned that the Values object is not available in the Chart.yaml file. Is there another way to accomplish this? In any case the --app-version flag seems logical to me considering the appVersion field for a Chart/Release is purely informational

**Pull request merged**